frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx

```diff
@@ -48,51 +59,59 @@ export const FormBuilder = ({
-                    fields={builderFields}
-                    isDraggingOver={snapshot.isDraggingOver}
-                  />
-                  {provided.placeholder}
-                  <BuilderAndDesignPlaceholder
-                    placeholderProps={placeholderProps}
+          {builderFields === null ? (
+            <BuilderFieldsSkeleton />
+          ) : (
+            <Droppable droppableId={FIELD_LIST_DROP_ID}>
+              {(provided, snapshot) =>
+                builderFields.length ? (
+                  <Box
+                    pos="relative"
+                    ref={provided.innerRef}
+                    {...provided.droppableProps}
+                  >
+                    <BuilderFields
+                      fields={builderFields}
+                      isDraggingOver={snapshot.isDraggingOver}
+                    />
+                    {provided.placeholder}
+                    <BuilderAndDesignPlaceholder
+                      placeholderProps={placeholderProps}
+                      isDraggingOver={snapshot.isDraggingOver}
+                    />
+                  </Box>
+                ) : (
+                  <EmptyFormPlaceholder
+                    m={{ base: '1.5rem', md: 0 }}
+                    ref={provided.innerRef}
+                    {...provided.droppableProps}
                     isDraggingOver={snapshot.isDraggingOver}
+                    onClick={handleBuilderClick}
                   />
-                </Box>
-              ) : (
-                <EmptyFormPlaceholder
-                  m={{ base: '1.5rem', md: 0 }}
-                  ref={provided.innerRef}
-                  {...provided.droppableProps}
-                  isDraggingOver={snapshot.isDraggingOver}
-                  onClick={handleBuilderClick}
-                />
-              )
-            }
-          </Droppable>
+                )
+              }
+            </Droppable>
+          )}
         </Flex>
-        <Button
-          _hover={{ bg: 'primary.200' }}
-          py="1.5rem"
-          mt="1.5rem"
-          variant="outline"
-          borderColor="secondary.200"
-          colorScheme="secondary"
-          height="auto"
-          onClick={() => {
-            setEditEndPage()
-            handleBuilderClick()
-          }}
-        >
-          <Text textStyle="subhead-2">Customise Thank you page</Text>
-        </Button>
+        {isLoading ? (
+          <Skeleton mt="1.5rem" h="4rem" />
+        ) : (
+          <Button
+            _hover={{ bg: 'primary.200' }}
+            py="1.5rem"
+            mt="1.5rem"
+            variant="outline"
+            borderColor="secondary.200"
+            colorScheme="secondary"
+            height="auto"
+            onClick={() => {
+              setEditEndPage()
+              handleBuilderClick()
+            }}
+          >
+            <Text textStyle="subhead-2">Customise Thank you page</Text>
+          </Button>
+        )}
```

Comment:
nit: could we wrap the button with a skeleton instead?

```tsx
<Skeleton isLoaded={!isLoading}>
  <Button
    ...
  </Button>
</Skeleton>
```

Reply:
for some reason the button ends up being half width when I do that. not too sure why that happens though!

Reply:
NVM i figured it out